doc/source_stream_library.tex

```diff
@@ -0,0 +1,206 @@
+\section*{Appendix: Stream library}
+
+Those stream library functions that are not built-ins are pre-declared as follows:
+
+\begin{lstlisting}
+// stream.js: Supporting streams in the Scheme style, following
+//            "stream discipline"
+// A stream is either null or a pair whose tail is
+// a nullary function that returns a stream.
+
+function list_to_stream(xs) {
+    return is_null(xs)
+        ? null
+        : pair(head(xs), 
+               () => list_to_stream(tail(xs))); 
+}
+
+// stream_to_list transforms a given stream to a list
+// Lazy? No: stream_to_list needs to force the whole stream
+function stream_to_list(xs) {
+    return is_null(xs)
+        ? null
+        : pair(head(xs), stream_to_list(stream_tail(xs)));
+}
+
+// stream_length returns the length of a given argument stream
+// throws an exception if the argument is not a stream
+// Lazy? No: The function needs to explore the whole stream
+function stream_length(xs) {
+    return is_null(xs)
+        ? 0
+        : 1 + stream_length(stream_tail(xs));
+}
+
+// stream_map applies first arg f to the elements of the second
+// argument, assumed to be a stream.
+// f is applied element-by-element:
+// stream_map(f,list_to_stream([1,[2,null]])) results in
```

Comment:
Refer to analogous list version.

Reply:
I don't see the mistake here.

Reply:
I'm referring to line 38, `stream_map(f,list_to_stream([1,[2,null]])) results in`, in particular `[1,[2,null]]`.